app/models/session.rb

```diff
@@ -32,6 +42,22 @@ def secure_random_token(length = DEFAULT_TOKEN_LENGTH)
   end
 
   def setup_defaults
-    @token ||= secure_random_token unless persisted?
+    # is this an existing old session without :created_at?
+    session_is_old = @created_at.nil? && persisted?
+
+    @token ||= secure_random_token
+    @created_at ||= Time.now.utc
+
+    # sessions only get saved at creation time.  For an existing session with a nil :created_at,
+    # we must forcibly re-save to redis else :created_at will always be nil and the session could
+    # theoretically last forever.  After being deployed 12 hours, this logic can be deleted then re-deployed
+    save if session_is_old
```

Comment:
I can't decide the best approach for migrating old Sessions (that dont have `created_at`) to the new.  As I have it now, it will migrate during run time in a just in time manner.  The logic is temporary and I'll have to remember to remove it.  Open to different opinions.